### PR TITLE
feat(github): Add GitHub Search to Github MCP

### PR DESCRIPTION
**Pull request by 0xRaduan**
## Description
Added comprehensive search functionality to the GitHub MCP Server, enabling users to search for code, issues/PRs, and users across GitHub. This enhancement brings powerful GitHub search capabilities directly to MCP clients.

## Server Details
- Server: `github`
- Changes to: `tools` (added search functionality), `schemas` (added search schemas), `documentation`

## Motivation and Context
GitHub's search functionality is a crucial feature for developers and AI assistants working with codebases. This addition allows MCP clients to:
- Search code across repositories with language and path filters
- Find relevant issues and pull requests
- Discover users and organizations
- Use GitHub's advanced search syntax
- Sort and paginate results efficiently

## How Has This Been Tested?
- Tested with Claude Desktop client
CleanShot 2024-11-28 at 09 56 24@2x
- Verified search functionality works through stdio interface
- Tested various search scenarios:
  - Code search with language and path filters
  - Issues search with labels and state filters
  - User search with location and follower filters

## Breaking Changes
No breaking changes. Users can continue using existing functionality without modification. The new search features are purely additive.

## Types of changes
- [ ] New MCP Server
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update

## Checklist
- [x] I have read the [MCP Protocol Documentation](https://modelcontextprotocol.io)
- [x] My server follows MCP security best practices
- [x] I have updated the server's README accordingly
- [x] I have tested this with an LLM client
- [x] My code follows the repository's style guidelines
- [ ] New and existing tests pass locally(there are no tests)
- [x] I have added appropriate error handling
- [x] I have documented all environment variables and configuration options

## Additional context
- Added comprehensive documentation for search query syntax in README
- Included detailed schema descriptions and JSDoc comments
- Implemented proper type safety with Zod schemas
- Version bumped from 0.5.1 to 0.6.0 for new feature addition

### CODE REVIEW

**Comment by 0xRaduan** [comment]
cc. @dsp-ant, couldn't assign a reviewer here, maybe you can help find the right person

File: src/github/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@modelcontextprotocol/server-github",
-  "version": "0.5.1",
+  "version": "0.6.0",
```
Comment (jspahrsummers) [inline src/github/package.json:3]: At the moment, all server versions have to be bumped in tandem. Can you please revert this or bump all servers?

File: src/github/schemas.ts [hunk 2]
```diff
@@ -285,94 +289,367 @@ export const GitHubPullRequestSchema = z.object({
...
+export const SearchCodeSchema = z.object({
+  q: z
+    .string()
+    .describe(
+      "Search query. See GitHub code search syntax: https://docs.github.com/en/search-github/searching-on-github/searching-code"
+    ),
+  sort: z
+    .enum(["", "indexed"])
+    .optional()
+    .describe("Sort field. Only 'indexed' is supported"),
```
Comment (jspahrsummers) [inline src/github/schemas.ts:543]: If only `indexed` is supported, seems like we should just remove this?

**Review by jspahrsummers — DISMISSED** [review]
Thank you for submitting this!

The diff is a bit hard to read with all the formatting changes (please strip those out in future), but the semantic changes look good. Please update from `main` so we can merge this.

**Comment by 0xRaduan** [comment]
Hey, taking a look right now.

File: src/github/schemas.ts (hunk 2 above)
Comment (0xRaduan) [reply src/github/schemas.ts:543]: agreed, will remove.

**Comment by 0xRaduan** [comment]
Re: formatting changes, I do think that we need to add a .prettier config, otherwise my VS Code was just setup by default differently, perhaps.

File: src/github/schemas.ts (hunk 2 above)
Comment (0xRaduan) [reply src/github/schemas.ts:543]: done.

**Comment by 0xRaduan** [comment]
Hey @jspahrsummers - I did address your comments + merged with latest master(as there were merge conflicts).

Feel free to squash and merge.

I did notice that there is no prettierc, so maybe I can add it in a separate PR.

Do let me know, if adding some CI for codestyle for TS/python makes sense? 

I think as you use UV for python, using ruff would make sense. As for TS, I think using prettier/eslint would make most sense.

**Review by jspahrsummers — DISMISSED** [review]
Thanks for making those changes!

w.r.t. linting and styling, I opened https://github.com/modelcontextprotocol/servers/issues/197 to track this. More than happy to review a PR addressing it!

**Pull request merged**